javalib/src/main/scala/java/lang/String.scala

```diff
@@ -1,42 +1,890 @@
 package java.lang
 
-import java.io.Serializable
-
-class _String private() extends CharSequence with Serializable with Comparable[_String] {
-  def this(value: Array[scala.Char], offset: scala.Int, count: scala.Int) = this()
-
-  def length(): scala.Int = ???
-  def charAt(index: scala.Int): Char = ???
-  def compareTo(another_String: _String): scala.Int = ???
-  def subSequence(beginIndex: scala.Int, endIndex: scala.Int): CharSequence = ???
-  def concat(str: _String): _String = ???
-  def indexOf(ch: scala.Int): scala.Int = ???
-  def indexOf(ch: scala.Int, fromIndex: scala.Int): scala.Int = ???
-  def lastIndexOf(ch: scala.Int): scala.Int = ???
-  def lastIndexOf(ch: scala.Int, fromIndex: scala.Int): scala.Int = ???
-  def substring(beginIndex: scala.Int): _String = ???
-  def substring(beginIndex: scala.Int, endIndex: scala.Int): _String = ???
-  def startsWith(prefix: _String): scala.Boolean = ???
-  def toLowerCase(): _String = ???
-  def toCharArray(): Array[Char] = ???
-  def equalsIgnoreCase(str: _String): scala.Boolean = ???
-  def getChars(srcBegin: scala.Int, srcEnd: scala.Int, dst: Array[scala.Char],
-      dstBegin: scala.Int): Unit = ???
-  override def hashCode(): scala.Int = ???
-
-  // TODO: rest of the api
+import java.io.{Serializable, UnsupportedEncodingException}
+import java.util._
+import java.util.regex._
+import java.nio._
+import java.nio.charset._
+
+final class _String private ()
```

Comment:
The 0-argument constructor should be public. It creates a string of length 0 (i.e., `""`).